src/middleware/src/storage/src/indexeddb.js

```diff
@@ -273,28 +273,31 @@ export default class IndexedDB {
     });
   }
 
-  static isSupported() {
-    const name = 'testIndexedDBSupport';
+  static loadAdapter(name) {
     const indexedDB = global.indexedDB || global.webkitIndexedDB || global.mozIndexedDB || global.msIndexedDB;
+    const db = new IndexedDB(name);
 
-    if (typeof indexedDB === 'undefined') {
-      return Promise.resolve(false);
+    if (typeof isSupported !== 'undefined') {
+      if (isSupported) {
+        return Promise.resolve(db);
+      }
+
+      return Promise.resolve(undefined);
     }
 
-    if (typeof isSupported !== 'undefined') {
-      return Promise.resolve(isSupported);
+    if (typeof indexedDB === 'undefined') {
+      isSupported = false;
+      return Promise.resolve(undefined);
     }
 
-    const db = new IndexedDB(name);
-    return db.save(name, { _id: '1' })
-      .then(() => db.clear())
+    return db.save('__testSupport', { _id: '1' })
```

Comment:
Yes it is possible you could see 2 DB entities. But in order to not show the popup we can't open 2 DB entries. This instead opens the DB used to store application data to also test support for the adapter.